### PR TITLE
GH-35 randomize validation time

### PR DESCRIPTION
**Pull request by makslion**
Randomize validation reconcilation delays

### CODE REVIEW

File: internal/controller/dnsrecord_controller.go [hunk 1]
```diff
@@ -69,6 +70,13 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	logger := log.FromContext(ctx)
 
 	reconcileStart = metav1.Now()
+
+	// randomize validation reconcile delay
+	seconds := float64(validationRequeueTime.Milliseconds())
```
Comment (maleck13) [inline internal/controller/dnsrecord_controller.go:75]: nit: is this var name a bit misleading? should it be milliseconds?

File: internal/controller/dnsrecord_controller.go [hunk 2]
```diff
@@ -69,6 +70,13 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+	validationRequeueTime = time.Millisecond * time.Duration(rand.Int63nRange(
+		int64(seconds*0.9),
+		int64(seconds*1.1)))
```
Comment (maleck13) [inline internal/controller/dnsrecord_controller.go:78]: ok so the interval will be between 90 and 110 % of the `validationRequeueTime`? wondering if we should increase the possible range a bit. Like 50-150% . Anyway just a thought

Comment (maleck13) [reply internal/controller/dnsrecord_controller.go:78]: possibly worth isolating into a function and adding a simple unit test?

**Review by maleck13 — APPROVED** [review]
looks ok to merge. just a couple small things to possibly improve

File: internal/controller/dnsrecord_controller.go (hunk 1 above)
Comment (makslion) [reply internal/controller/dnsrecord_controller.go]: renamed

**Review by maleck13 — APPROVED** [review]

File: internal/controller/dnsrecord_controller.go (hunk 2 above)
Comment (makslion) [reply internal/controller/dnsrecord_controller.go]: increased variance to 50% and isolated into a separate function. 

Not sure though how to reliably test the randomization function: we could have a test just ensuring that we get a valid value from this (but given how the code works we always will have a valid `time.Duration` value out of this) and we already testing for "usable" duration in the integration suite (if it to randomize seconds into minutes we will timeout out of the integration suite). The suggested way to test randomness is through a lot of iterations to build up a statistically significant amount of data - that doesn't sound like something worth doing. 
In any case - I put together a tiny test that should be what you had in mind 

Went through the package and there are a few functions that could use unit tests as well. If we do this - maybe better to move our integration suite into a separate package and only keep `filename_test.go` for the unit suite? I can create a issue for "better days" to reorganize tests in the DNS Operator if that sounds cool :thinking:

Comment (maleck13) [reply internal/controller/dnsrecord_controller.go]: Yeah not huge value here, I guess we could do a number of iterations and ensure the value is within the expected range?

File: internal/common/helper_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+func TestRandomizeDuration(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		variance float64
+		duration time.Duration
+	}{
+		{
+			name:     "returns valid duration in range",
```
Comment (maleck13) [inline internal/common/helper_test.go:16]: yep this is fine perhaps forcing it to do a loop and  a set number of iterations testing each is within the range?

**Review by maleck13 — APPROVED** [review]

File: internal/common/helper_test.go (hunk 3 above)
Comment (makslion) [reply internal/common/helper_test.go:17]: :facepalm: yep, that is smart. Done

**Pull request merged**